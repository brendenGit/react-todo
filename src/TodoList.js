import { useState } from "react";
import { v4 as uuid } from 'uuid';

import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm.js";

function TodoList() {
    const [todos, setTodos] = useState([]);

    const addTodo = ({ text }) => {
        console.log(text);
        const newTodo = { text, id: uuid() }
        setTodos(oldTodos => ([...oldTodos, newTodo]));
    };
 
    const removeTodo = (id) => {
        setTodos(oldTodos => (oldTodos.filter(todo => todo.id !== id)));
    };

    return (
        <div>
            <NewTodoForm addTodo={addTodo} />
            <ul>
                {todos.map(todo => <Todo 
                                    text={todo.text} 
                                    id={todo.id}
                                    removeTodo={removeTodo}
                                />)}
            </ul>
        </div>
    )
};

export default TodoList;