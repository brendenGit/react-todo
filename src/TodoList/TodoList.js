import { useState } from "react";
import { v4 as uuid } from 'uuid';

import Todo from "../Todo/Todo.js";
import NewTodoForm from "../NewTodoForm/NewTodoForm.js";

function TodoList() {
    const [todos, setTodos] = useState([]);

    const addTodo = ({ text }) => {
        const newTodo = { text, id: uuid() }
        setTodos(oldTodos => ([...oldTodos, newTodo]));
    };
 
    const removeTodo = (id) => {
        setTodos(oldTodos => (oldTodos.filter(todo => todo.id !== id)));
    };

    return (
        <div>
            <NewTodoForm addTodo={addTodo} />
            <ul>
                {todos.map(todo => <Todo 
                                    text={todo.text} 
                                    id={todo.id}
                                    key={todo.id}
                                    removeTodo={removeTodo}
                                />)}
            </ul>
        </div>
    )
};

export default TodoList;