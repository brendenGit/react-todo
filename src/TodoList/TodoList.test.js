import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

//Smoke test
it("renders without crashing", function () {
    render(<TodoList />)
});

//Snapshot test
it("matches snapshot", function () {
    const { asFragment } = render(<TodoList />)
    expect(asFragment()).toMatchSnapshot();
});

//Specialized testing below
test('adds and removes todos', () => {
    const { getByLabelText, getByText, getByTestId } = render(<TodoList />);

    // Add a todo
    fireEvent.change(getByLabelText('Text:'), { target: { value: 'Buy groceries' } });
    fireEvent.click(getByText('Submit'));

    // Check if the todo is added
    const addedTodo = getByText('Buy groceries');
    expect(addedTodo).toBeInTheDocument();

    // Remove the added todo
    fireEvent.click(getByText('X'));

    // Check if the todo is removed
    expect(addedTodo).not.toBeInTheDocument();
});