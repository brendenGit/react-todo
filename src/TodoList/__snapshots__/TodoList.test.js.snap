// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="text"
      >
        Text:
        <textarea
          id="text"
          name="text"
          type="text"
        />
      </label>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
